stackslib/src/main.rs

```diff
@@ -974,6 +1142,28 @@ simulating a miner.
         process::exit(1);
     }
 
+    if argv[1] == "getnakamotoinv" {
+        let peer_addr: SocketAddr = argv[2].to_socket_addrs().unwrap().next().unwrap();
+        let data_port: u16 = argv[3].parse().unwrap();
+        let ch = ConsensusHash::from_hex(&argv[4]).unwrap();
+
+        let mut session = P2PSession::begin(peer_addr, data_port).unwrap();
+
+        // send getnakamotoinv
+        let get_nakamoto_inv =
+            StacksMessageType::GetNakamotoInv(GetNakamotoInvData { consensus_hash: ch });
+
+        let msg = session.make_peer_message(get_nakamoto_inv).unwrap();
+        session.send_peer_message(msg).unwrap();
+        let resp = session.recv_peer_message().unwrap();
+
+        let StacksMessageType::NakamotoInv(inv) = &resp.payload else {
+            panic!("Got spurious message: {:?}", &resp);
```

Comment:
When I run this, targeted at my naka node, I get:

```
Got spurious message: StacksMessage { preamble: Preamble { peer_version: 4207599114, network_id: 2147483648, seq: 4167833110, burn_block_height: 75190, burn_block_hash: 3e6d9605bdddaa22b6497ee6e518bdea631512c76180a277cdcdb9d3b25b94f1, burn_stable_block_height: 75189, burn_stable_block_hash: 69accdd0003b8f6c8d522066e3f556a7a00fb0af206c9140e42a1c746ad4f89e, additional_data: 0, signature: 0035998ea0a3a63681150f5788ee02f3e68ef76b1a3520d1dbb869babe4bf462612838b6f84d43693274f4af54a4d489d53806b56ed064896bd6aecf5cb541672e, payload_len: 269 }, relayers: [RelayData { peer: e728418423378cf56244712f0bb5983b2d84f1cf://34.150.184.50:20444, seq: 0 }, RelayData { peer: 98d81b3c2c991fe51710a7ea329e4ae1e77f8068://34.150.184.50:20444, seq: 0 }], payload: Transaction(StacksTransaction { version: Testnet, chain_id: 2147483648, auth: Standard(Singlesig(SinglesigSpendingCondition { hash_mode: P2PKH, signer: e959844d85de7954970675332446df1c5a54419c, nonce: 8652, tx_fee: 300, key_encoding: Compressed, signature: 009821ecf87a26b5662cba9112ff5f3a95799533301d6f4886079042ef8471af0e3860f4655f383ee9ebddff144eb6db408354ecf53f21e3133faf4fe5e8ac7d90 })), anchor_mode: Any, post_condition_mode: Deny, post_conditions: [], payload: TokenTransfer(Standard(StandardPrincipalData(ST484MS3VACPAZ90WHC21XQ7T6XANCV341HJYE0W)), 1000, 00000000000000000000000000000000000000000000000000000000000000000000) }) }
```

Is this expected? Note that I just used the `pox_consensus` I got from `/v2/info`.

Reply:
Yes, this is currently expected.  This PR also fixes this -- nodes accidentally broadcast messages to unauthenticated peers (which causes this).